crates/std_detect/src/detect/os/windows/aarch64.rs

```diff
@@ -0,0 +1,42 @@
+//! Run-time feature detection for Aarch64 on Windows.
+
+use crate::detect::{cache, Feature};
+
+/// Try to read the features from OS API.
+pub(crate) fn detect_features() -> cache::Initializer {
+    const PF_ARM_NEON_INSTRUCTIONS_AVAILABLE: u32 = 19;
+    const PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE: u32 = 30;
+    const PF_ARM_V8_CRC32_INSTRUCTIONS_AVAILABLE: u32 = 31;
+
+    extern "system" {
+        pub fn IsProcessorFeaturePresent(ProcessorFeature: u32) -> u32;
+    }
+
+    let mut value = cache::Initializer::default();
+    {
+        let mut enable_feature = |f, enable| {
+            if enable {
+                value.set(f as u32);
+            }
+        };
+
+        // Some features such Feature::fp may be supported on current CPU,
+        // but no way to detect it by OS API.
+
+        enable_feature(Feature::asimd, unsafe {
```

Comment:
It's ok to put all of these inside a single `unsafe { ... }` block to avoid repeating unsafe - just add a comment stating that `unsafe` is required for the extern "system" calls.